### PR TITLE
Bug 1224097: Don't use custom names for route.servicename

### PR DESCRIPTION
**Pull request by kargakis**

### CODE REVIEW

**Comment by kargakis** [comment]
@smarterclayton ptal

**Comment by kargakis** [comment]
@bparees fyi

**Comment by bparees** [comment]
hmm..so i can't create multiple routes for a service?

**Comment by kargakis** [comment]
> hmm..so i can't create multiple routes for a service?

One route atm. Come to think of it, we never discussed anything about multiple routes.

**Comment by smarterclayton** [comment]
It's an optional thing, most people want one route.  We'd probably just want to support a way to optionally pass a different name (and when a route exists and you don't specify that we can say "route already exists")

**Comment by kargakis** [comment]
Can I get this merged?

File: pkg/cmd/cli/cmd/expose.go [hunk 1]
```diff
@@ -13,15 +13,20 @@ import (
 )
 
 const (
-	exposeLong = `Take a replicated application and expose it as a route.
+	exposeLong = `Look up a service and expose it as a route.
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/expose.go]: I would change this to "Expose containers internally as services or externally via routes"

File: pkg/cmd/cli/cmd/expose.go [hunk 2]
```diff
@@ -46,6 +51,14 @@ func validate(cmd *cobra.Command, f *clientcmd.Factory, args []string) error {
 	if cmdutil.GetFlagString(cmd, "generator") != "route/v1" {
 		return nil
 	}
+	// We don't want to use any other name for our route other than
+	// the one the service we expose has
+	providedName := cmdutil.GetFlagString(cmd, "name")
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/expose.go]: Why does a route have to have the name of the service it exposes?

Comment (kargakis) [inline pkg/cmd/cli/cmd/expose.go]: Currently there is no other way to pass parameters into the route generation other than a [map](https://github.com/openshift/origin/blob/f11daef358df0f3424acb9e3cf39cafacfbff369/pkg/route/generator/generate.go#L16). These parameters are being set up by expose's flags and there is only one *name* parameter we can pass.:)

Comment (smarterclayton) [inline pkg/cmd/cli/cmd/expose.go]: Hrm.  It's not flexible like I was hoping.

Comment (kargakis) [inline pkg/cmd/cli/cmd/expose.go]: If we can add our own flags on top then it's going to be trivial to add parameters in the generator. Gonna play around with it on Monday.

**Comment by liggitt** [comment]
A TLS and non TLS route seems pretty likely

**Comment by kargakis** [comment]
@smarterclayton I've added the `--route-name` flag and now we can specify our own route names:)

**Comment by smarterclayton** [comment]
Well, we probably want both to be the same route in the long term.  But in the short term yes.

**Comment by smarterclayton** [comment]
Part of the problem here is that the original object is not available to us - instead, we're trying to make Expose do something it's not intended to do.  I don't think it's necessary that we have to add our own parameter just to make this work - instead we should be able to get both the input object name and the provided name arg.  If the original input name of the object was always passed as "original-name" that would allow us to avoid needing to check route name.

**Comment by kargakis** [comment]
@smarterclayton can we merge this now that it's in upstream?

**Comment by smarterclayton** [comment]
LGTM [merge] thanks for shepherding this

**Pull request merged**